new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Kubernetes.java

```diff
@@ -1327,5 +1352,149 @@ public static boolean deleteClusterRoleBinding(String name) {
     return true;
   }
 
+  // --------------------------- Exec   ---------------------------
+
+  /**
+   * Execute a command in a container.
+   *
+   * @param pod The pod where the command is run
+   * @param containerName The container in the Pod where the command is run. If no container
+   *     name is provided than the first container in the Pod is used.
+   * @param redirectToStdout copy Process output to stdout
+   * @param command The command to run
+   * @return result of command execution
+   * @throws IOException if an I/O error occurs.
+   * @throws ApiException if Kubernetes client API call fails
+   * @throws InterruptedException if any thread has interrupted the current thread
+   */
+  public static ExecResult exec(V1Pod pod, String containerName, boolean redirectToStdout,
+      String... command)
+      throws IOException, ApiException, InterruptedException {
+
+    // Execute command using Kubernetes API
+    KubernetesExec kubernetesExec = createKubernetesExec(pod, containerName);
+    final Process proc = kubernetesExec.exec(command);
+
+    final CopyingOutputStream copyOut =
+        redirectToStdout ? new CopyingOutputStream(System.out) : new CopyingOutputStream(null);
```

Comment:
It seems that if the redirectToStdout is false, the following code will still make a copy of the original  input stream while the original input stream can be directly used.

Reply:
Instead of waiting for Process.waitFor() to read from the stream directly, I wanted to creating a separate thread (used for redirect) in order to consume the process's output stream to prevent the output stream buffer from becoming full and blocking.